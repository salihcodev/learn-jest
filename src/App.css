main.to-do-wrapper {
  transition: all ease-in-out 0.5s;
}
main.to-do-wrapper .app-container {
  width: 40rem;
  margin: 8rem auto;
  min-height: 20rem;
  padding: 1rem;
  border-radius: 0.7rem;
  box-shadow: black 4px 8px 0px 10px;
}
