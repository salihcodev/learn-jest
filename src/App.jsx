import React from 'react';
import './App.css';
import AddingTodosForm from './components/add-item-form/AddingTodosForm.comp';
import TodoList from './components/todo-list/TodoList.comp';

function App({ appTitle }) {
  return (
    <main className="to-do-wrapper">
      <div className="app-container">
        <h2 className="wrapper-heading">{appTitle}</h2>
        {/* form control */}
        <AddingTodosForm />
        <TodoList />
      </div>
    </main>
  );
}

export default App;
