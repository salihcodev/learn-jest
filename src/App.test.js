import React from 'react';
import App from './App';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

describe("Check for 'App'", () => {
  let component;
  let props;

  Enzyme.configure({
    adapter: new Adapter(),
  });

  it("Should have 'appTitle' prop", () => {
    props = {
      appTitle: 'Add todo item.',
    };
    component = shallow(<App appTitle={props.appTitle} />);

    expect(component).toMatchSnapshot();
  });
});
