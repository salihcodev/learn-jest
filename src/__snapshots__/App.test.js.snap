// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for 'App' Should have 'appTitle' prop 1`] = `ShallowWrapper {}`;
